org.eclipse.scanning.points/src/org/eclipse/scanning/points/CollatedStepGenerator.java

```diff
@@ -12,9 +12,14 @@
 package org.eclipse.scanning.points;
 
 import org.eclipse.scanning.api.ModelValidationException;
+import org.eclipse.scanning.api.points.ScanPointIterator;
 import org.eclipse.scanning.api.points.models.CollatedStepModel;
 import org.eclipse.scanning.api.points.models.StepModel;
 
+/**
+ * An iterator along points where one or more axes move to the same value together.
+ * TODO: this class should be reimplemented in jython or removed.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)